src/routes/roomMusic.routes.ts

```
@@ -116,4 +85,50 @@ roomMusicRouter.get('/song-info/:videoId', async (req, res, next) => {
+ */
+roomMusicRouter.get('/:roomId/search-songs', async (req, res) => {
+  const { q, limit = '50' } = req.query
+  const parsedLimit = parseInt(limit as string, 10)
+
+  // Validate search query
+  if (!q || typeof q !== 'string') {
+    return res.status(400).json({ error: 'Missing or invalid search query' })
+  }
+
+  // Validate limit parameter
+  if (isNaN(parsedLimit) || parsedLimit < 1 || parsedLimit > 100) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)